samcli/local/apigw/local_apigw_service.py

```diff
@@ -397,6 +403,78 @@ def _parse_lambda_output(lambda_output, binary_types, flask_request):
 
         return status_code, headers, body
 
+    @staticmethod
+    def _parse_v2_payload_format_lambda_output(lambda_output, binary_types, flask_request):
+        """
+        Parses the output from the Lambda Container
+
+        :param str lambda_output: Output from Lambda Invoke
+        :return: Tuple(int, dict, str, bool)
+        """
+        # pylint: disable-msg=too-many-statements
+        try:
+            json_output = json.loads(lambda_output)
+        except ValueError as ex:
+            raise LambdaResponseParseException("Lambda response must be valid json") from ex
+
+        # lambda can return any valid json response in payload format version 2.0.
+        # response can be a simple type like string, or integer
+        # https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.response
+        if isinstance(json_output, dict):
+            body = json_output.get("body") if "statusCode" in json_output else json.dumps(json_output)
+        else:
+            body = json_output
+            json_output = {}
+
+        if not body:
+            LOG.warning("Lambda returned empty body!")
+
+        status_code = json_output.get("statusCode") or 200
+        headers = LocalApigwService._merge_response_headers(
+            json_output.get("headers") or {}, json_output.get("multiValueHeaders") or {}
+        )
+
+        is_base_64_encoded = json_output.get("isBase64Encoded") or False
+
+        try:
+            status_code = int(status_code)
+            if status_code <= 0:
+                raise ValueError
+        except ValueError as ex:
+            raise LambdaResponseParseException("statusCode must be a positive int") from ex
+
+        try:
+            if body:
+                body = str(body)
+        except ValueError as ex:
+            raise LambdaResponseParseException(
+                f"Non null response bodies should be able to convert to string: {body}"
+            ) from ex
+
+        # API Gateway only accepts statusCode, body, headers, and isBase64Encoded in
+        # a response shape.
+        # Don't check the response keys when inferring a response, see
+        # https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.v2.
+        if "statusCode" in json_output:
+            invalid_keys = LocalApigwService._invalid_apig_response_keys(json_output)
+            if bool(invalid_keys):
+                raise LambdaResponseParseException(
+                    f"Invalid API Gateway Response Keys: {invalid_keys} in {json_output}"
+                )
+
+        # If the customer doesn't define Content-Type default to application/json
+        if "Content-Type" not in headers:
```

Comment:
MIME types are [case insensitive](https://tools.ietf.org/html/rfc2045); e.g. `content-type` is a valid header but won't work here. Are we normalizing the headers before this?

Reply:
headers object is of type werkzeug.datastructures.Headers .. which ignore the key case while checking if the key is exist or not.